pygmt/helpers/decorators.py

```diff
@@ -5,14 +5,16 @@
 arguments, insert common text into docstrings, transform arguments to strings,
 etc.
 """
-import textwrap
 import functools
+import inspect
+import os
+import textwrap
 
 import numpy as np
+from matplotlib.testing.compare import compare_images
```

Comment:
```
from matplotlib.testing.compare import compare_images
``` 
As I understand it, the code means that now `matplotlib` becomes a required dependency, even for users who never run the tests, right? 

Although PyGMT already requires `matplotlib` for testings and most users usually have matplotlib installed. I still don't want to add one dependency to PyGMT. 

When I wrote the first commit (8b7861441f3a150dbc01fba4465ae577db766328), I put the codes in `pygmt/helpers/testing.py`. By doing that way, I think matplotlib is still optional, although I haven't tested it.

Reply:
Good point, I think you're right here. I also encountered issues with circular imports when moving the code to decorators.py, hence this line:

https://github.com/GenericMappingTools/pygmt/blob/04b3f4154ab35a345c3394cd393e050f306f64ae/pygmt/helpers/decorators.py#L459

Probably should move it back under `pygmt/helpers/testing.py` then. As an aside, I've opened up a feature request at https://github.com/matplotlib/pytest-mpl/issues/94, and we might be able to do all this from `pytest-mpl` in the future.